### PR TITLE
[dcl.init.aggr]/1 Aggregate only cares about direct base classes CWG2610

### PR DESCRIPTION
**Pull request by languagelawyer**
Fixes https://github.com/cplusplus/CWG/issues/101
See also https://github.com/cplusplus/draft/pull/2892

### CODE REVIEW

**Review by jensmaurer — CHANGES_REQUESTED** [review]
I think indirect virtual base classes may be a problem (they have global constructor effects). Agreed with the private/protected changes.

File: source/declarations.tex [hunk 1]
```diff
@@ -4809,7 +4809,7 @@
 \item
 no virtual functions\iref{class.virtual}, and
 \item
-no virtual, private, or protected base classes\iref{class.mi}.
+no virtual, direct private, or direct protected base classes\iref{class.mi}.
```
Comment (hubert-reinterpretcast) [inline source/declarations.tex:4812]: Perhaps better to split this into two bullets?

Comment (languagelawyer) [reply source/declarations.tex:4812]: Or "or base classes" can be added to the previous bullet, and "virtual" removed from this one?

Comment (hubert-reinterpretcast) [reply source/declarations.tex:4812]: > Or "or base classes" can be added to the previous bullet, and "virtual" removed from this one?

Sounds good.

**Review by jensmaurer — APPROVED** [review]

**Comment by jensmaurer** [comment]
This looks good to me now.

@hubert-reinterpretcast , please have a final look.

**Comment by hubert-reinterpretcast** [comment]
> This looks good to me now.
> 
> @hubert-reinterpretcast , please have a final look.

It's good enough. I have a mild preference for repeating "virtual" and reordering the bullets so that the two "no private or protected" bullets are grouped together.

**Review by hubert-reinterpretcast — APPROVED** [review]
LGTM; thanks!

**Comment by tkoeppe** [comment]
@jensmaurer Does CWG consider this editorial? (Feel free to merge if so!)

**Comment by jensmaurer** [comment]
CWG2610

**Pull request closed without merging**